app/controllers/markets_controller.rb

```diff
@@ -1,2 +1,58 @@
 class MarketsController < ApplicationController
+
+  def index
+    @markets = Market.all
+  end
+
+  def new
+    @market = Market.new
+  end
+
+  def create
+    Market.create(market_params[:market])
+    redirect_to "/markets"
+  end
+
+  def edit
+    id = params[:id]
+    @market = Market.find(id)
+  end
+
+  def show
+    id = params[:id]
+    @market = Market.find(id)
+    @vendors = Market.find(id).vendors
+  end
+
+  def update
+    id = params[:id]
+    @market = Market.find(id)
```

Comment:
It would be great to see some refactoring of the `id` and `@market` variables to use a `before_action`